MAINT: Remove dead code backporting py2.6 warnings

eric-wieser opened the pull request:
Since this is now in `np.testing._private`, it's no longer usable by the outside world anyway

eric-wieser committed:
MAINT: Remove dead code backporting py2.6 warnings

Since this is now in `np.testing._private`, it's no longer usable by the outside world anyway

charris commented on numpy/testing/_private/utils.py:
```diff
@@ -1639,98 +1639,6 @@ def integer_repr(x):
 
-# The following two classes are copied from python 2.6 warnings module (context
-# manager)
-class WarningMessage(object):
```
Need to import this from `warnings`.

charris replied:
Did that, waiting on tests.

eric-wieser replied:
Thanks, guess my search for uses missed that one

charris committed:
BUG: Import WarningMessage from warnings.

charris commented:
Should probably add a release note on the removals.

eric-wieser commented:
@charris: A brief history:

* 1.13: Available at `np.testing.utils.WarningManager`
* 1.14: Moved to `np.testing.nose_tools.utils.WarningManager`
* 1.15-dev: Moved to `np.testing._private.utils`
* This patch: removed

It seems that the release note was due in 1.14, and that ship has sailed.

I don't have the full picture for what's going on with these moves - since I think you made them, can I ask you adjust the release notes however you see fit?

charris commented:
The imports use `numpy.testing`,  but I see it only does `from ._private.utils import *`, so those classes were missing anyway. Imports from `numpy.testing.utils` are deprecated.

charris commented:
Thanks Eric.
